### PR TITLE
eventhandler: Refactor tests

### PR DESCRIPTION
**Pull request by samuelkarp**
### Summary
Fixes https://github.com/aws/amazon-ecs-agent/issues/490
Changes a single 300-line test into 6 separate tests.

### Implementation details
Removed hand-rolled mock in favor of a gomock mock.  Broke test cases out into separate tests.

### Testing
- [x] Builds (`make release`)
- [x] Unit tests (`make short-test`) pass
- [x] Integration tests (`make test`) pass
- [x] Functional tests (`make run-functional-tests`) pass

New tests cover the changes: yes (it's all tests)

### Description for the changelog
No changelog impact

### Licensing
This contribution is under the terms of the Apache 2.0 License: yes (Amazon employee)

r? @aaithal @kiranmeduri @richardpen @nmeyerhans

### CODE REVIEW

File: agent/eventhandler/handler_test.go [hunk 1]
```diff
@@ -16,185 +16,84 @@ package eventhandler
...
+	client := mock_api.NewMockECSClient(ctrl)
 
 	// Trivial: one container, no errors
+	contCalled := make(chan struct{})
+	taskCalled := make(chan struct{})
+	contEvent1 := contEvent("1")
+	contEvent2 := contEvent("2")
+	taskEvent2 := taskEvent("2")
+
+	client.EXPECT().SubmitContainerStateChange(contEvent1).Do(func(interface{}) { contCalled <- struct{}{} })
```
Comment (aaithal) [inline agent/eventhandler/handler_test.go]: can you wrap these (and other such instances in this file) in a `gomock.InOrder()` block as well?

Comment (samuelkarp) [inline agent/eventhandler/handler_test.go]: The order of the two `SubmitContainerStateChange` calls is non-deterministic, so `gomock.InOrder` fails due to that.

Comment (aaithal) [inline agent/eventhandler/handler_test.go]: ah, alright. :shipit:

File: agent/eventhandler/handler_test.go [hunk 2]
```diff
@@ -203,165 +102,99 @@ func TestSendsEvents(t *testing.T) {
 	}
```
Comment (richardpen) [inline agent/eventhandler/handler_test.go]: Can we use go routine here? line: 101

Comment (samuelkarp) [inline agent/eventhandler/handler_test.go]: We don't need it; `AddContainerEvent` already creates goroutines inside and has async behavior.

**Pull request merged**